com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/client/DatabaseQueryClient.java

```diff
@@ -0,0 +1,171 @@
+package com.sap.cloud.lm.sl.cf.database.migration.client;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.List;
+
+import com.sap.cloud.lm.sl.cf.database.migration.executor.type.DatabaseTypeSetter;
+import com.sap.cloud.lm.sl.cf.database.migration.executor.type.DatabaseTypeSetterFactory;
+import com.sap.cloud.lm.sl.cf.database.migration.metadata.DatabaseTableColumnMetadata;
+import com.sap.cloud.lm.sl.cf.database.migration.metadata.DatabaseTableMetadata;
+import com.sap.cloud.lm.sl.cf.database.migration.metadata.ImmutableDatabaseTableColumnMetadata;
+import com.sap.cloud.lm.sl.cf.database.migration.metadata.ImmutableDatabaseTableMetadata;
+import com.sap.cloud.lm.sl.cf.persistence.query.SqlQuery;
+import com.sap.cloud.lm.sl.cf.persistence.util.JdbcUtil;
+import com.sap.cloud.lm.sl.cf.persistence.util.SqlQueryExecutor;
+
+public class DatabaseQueryClient {
+    private final SqlQueryExecutor sqlQueryExecutor;
+
+    public DatabaseQueryClient(SqlQueryExecutor sqlQueryExecutor) {
+        this.sqlQueryExecutor = sqlQueryExecutor;
+    }
+
+    public long getLastSequenceValue(String sequenceName) throws SQLException {
+        return executeWithAutoCommit((connection) -> {
```

Comment:
You don't need the braces around the lambda's parameter if you have just one parameter.